scripts/install_dict.py

```diff
@@ -0,0 +1,178 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""A script installing Hunspell dictionaries.
+
+Use: python -m scripts.install_dict [--list] [lang [lang [...]]]
+"""
+
+import argparse
+import base64
+import json
+import os
+import sys
+import re
+import urllib.parse
+import urllib.request
+import attr
+
+from qutebrowser.browser.webengine import spell
+from qutebrowser.config import configdata
+
+
+API_URL = 'https://chromium.googlesource.com/chromium/deps/hunspell_dictionaries.git/+/master/'
+
+
+@attr.s
+class Language:
+    """Dictionary language specs."""
+
+    code = attr.ib(None)
+    name = attr.ib(None)
+    file_basename = attr.ib(None)
+    file_extension = attr.ib('bdic')
+
+    @property
+    def file_path(self):
+        return '.'.join([self.file_basename, self.file_extension])
+
+
+def get_argparser():
+    """Get the argparse parser."""
+    desc = 'Install Hunspell dictionaries for QtWebEngine.'
+    parser = argparse.ArgumentParser(prog='install_dict',
+                                     description=desc)
+    parser.add_argument('-l', '--list', action='store_true',
+                        help="Display the list of available languages.")
+    parser.add_argument('languages', nargs='*',
+                        help="A list of languages to install.")
+    return parser
+
+
+def print_list(languages):
+    for lang in languages:
+        print('{1}\t{0}'.format(lang.name, lang.code))
+
+
+def valid_languages():
+    """Return a mapping from valid language codes to their names."""
+    option = configdata.DATA['spellcheck.languages']
+    return option.typ.valtype.valid_values.descriptions
+
+
+def language_list_from_api():
+    """Return a JSON with a list of available languages from Google API."""
+    listurl = urllib.parse.urljoin(API_URL, '?format=JSON')
+    response = urllib.request.urlopen(listurl)
+    # TODO: what's up with the first 4 characters?
+    entries = json.loads(response.read().decode('utf-8')[4:])['entries']
+    return entries
+
+
+def available_languages():
+    """Return a list of Language objects of all available languages."""
+    lang_map = valid_languages()
+    api_list = language_list_from_api()
+    dict_re = re.compile(r"""
+        (?P<filename>(?P<dict>[a-z]{2}(-[A-Z]{2})?).*)\.bdic
+    """, re.VERBOSE)
+    code2file = {}
+    for lang in api_list:
+        match = dict_re.match(lang['name'])
+        if match is not None:
+            code2file[match.group('dict')] = match.group('filename')
+    return [
+        Language(code, name, code2file[code])
+        for code, name in lang_map.items()
+    ]
+
+
+def download_dictionary(url, dest):
+    """Download a decoded dictionary file."""
+    response = urllib.request.urlopen(url)
+    decoded = base64.decodebytes(response.read())
```

Comment:
I really wish there was a way to just download a file from Google's repository browser without involving base64, but I couldn't find one... weird.